### PR TITLE
docs(headless): added react sample for smart snippet source analytics

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-1498

Split from: https://github.com/coveo/ui-kit/pull/1876

### CODE REVIEW

File: packages/samples/headless-react/src/components/smart-snippet/smart-snippet.class.tsx [hunk 1]
```diff
@@ -90,6 +91,23 @@ export class SmartSnippet extends Component<{}, SmartSnippetState> {
...
+                onMouseUp={() => this.controller.source?.select()}
+                onTouchStart={() =>
+                  this.controller.source?.beginDelayedSelect()
+                }
+                onTouchEnd={() => this.controller.source?.cancelPendingSelect()}
+              >
+                Source
+              </a>
+            ) : (
+              []
```
Comment (liviarett) [inline packages/samples/headless-react/src/components/smart-snippet/smart-snippet.class.tsx:109]: how come you return an empty array here?
isn't it cleaned to do 
`Boolean(source) && <Component />` ?

Comment (btaillon) [reply packages/samples/headless-react/src/components/smart-snippet/smart-snippet.class.tsx:109]: in a few instances, I saw `false` being rendered. So now, I'm avoiding using the `&&` pattern in JSX.

Comment (liviarett) [reply packages/samples/headless-react/src/components/smart-snippet/smart-snippet.class.tsx:107]: odd! I've seen 0 before, but never false.
fair enough

Comment (btaillon) [reply packages/samples/headless-react/src/components/smart-snippet/smart-snippet.class.tsx:107]: I may have seen it in attributes

Comment (olamothe) [reply packages/samples/headless-react/src/components/smart-snippet/smart-snippet.class.tsx:107]: I personally prefer extracting any logic from the JSX code, and just have a `renderSource()` function which itself contains the `if/else` code.

Just seems easier to read ?

**Review by olamothe — APPROVED** [review]

File: packages/samples/headless-react/src/components/smart-snippet/smart-snippet.class.tsx (hunk 1 above)
Comment (btaillon) [reply packages/samples/headless-react/src/components/smart-snippet/smart-snippet.class.tsx]: In Atomic that's generally what I do, I didn't do it here for the sake of having the whole thing in the same function, but I'll extract it.

**Review by ThibodeauJF — APPROVED** [review]

**Pull request merged**